parsec/parsec.c

```diff
@@ -477,44 +480,53 @@ parsec_context_t* parsec_init( int nb_cores, int* pargc, char** pargv[] )
     parsec_debug_init();
     mca_components_repository_init();
 
-    parsec_mca_param_reg_int_name("runtime", "warn_slow_binding", "Warn when the runtime detects poorly performing binding configuration, check distributed binding iff the number of nodes is lower than the parameter (0 to silence all warnings)", false, false, slow_bind_warning, &slow_bind_warning);
-
-#if defined(PARSEC_HAVE_HWLOC)
-    parsec_hwloc_init();
-#endif  /* defined(HWLOC) */
-
+    parsec_mca_param_reg_int_name("runtime", "warn_slow_binding",
+                                  "Warn when the runtime detects poorly performing binding configuration, check distributed binding iff the number of nodes is lower than the parameter (0 to silence all warnings)",
+                                  false, false, slow_bind_warning, &slow_bind_warning);
+    parsec_mca_param_reg_int_name("runtime", "show_bindings",
+                                  "Show the binding of all PaRSEC resources (main thread, threads and communication thread",
+                                  false, false, parsec_show_bindings, &parsec_show_bindings);
+    parsec_mca_param_reg_int_name("runtime", "ignore_bindings",
+                                  "Ignore the bindings provided by the process manager (batch scheduler or mpirun) and use all available resources on the node",
+                                  false, false, parsec_runtime_ignore_bindings, &parsec_runtime_ignore_bindings);
+    parsec_mca_param_reg_int_name("runtime", "num_cores", "The total number of cores to be used by the runtime (-1 for all available)",
+                                  false, false, parsec_runtime_max_number_of_cores, &parsec_runtime_max_number_of_cores);
+    parsec_mca_param_reg_int_name("runtime", "allow_pu",
+                                  "Allow threads to bind to PU (processing unit) instead of cores.",
```

Comment:
What's a PU? A socket?

Reply:
processing unit, a logical unit bound to a physical core. In general it is a core, except when there is hyper-threading when the PU is a hyper-thread and several hyperthreads make a core.